include/bst.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef __BINARY_SEARCH_TREE__
+#define __BINARY_SEARCH_TREE__
+
+template <class T>
+class BNode {
+public:
+    const T& operator*() const;
+    T& operator*();
+};
+
+template <typename T>
+class BinarySearchTree {
+private:
+    
+public:
+    typedef BNode* iterator;
+
+    BinarySearchTree();
+    ~BinarySearchTree();
+
+    BinarySearchTree(const BinarySearchTree &other);
+    BinarySearchTree &operator=(const BinarySearchTree &other);
+
+    void insert(const T &value);
+    size_t erase(const iterator it);
```

Comment:
`erase` method return type should be `bool` instead of `size_t` so it is not clear what should return it, e.g.: size? which is element removed? status?